File: examples/routines/function.f90
real function f(x)
    implicit none

    real :: x

    f = x*x

end function f

program function_example
    implicit none
    
    real :: x = 1.5
    real :: f;

    write(*,*) "Fortran function example"

    write(*,*) x, " squared is ", f(x)

end program function_example

File: examples/routines/function.cpp
#include<iostream>


double f(double x) {
    return x*x;
}

int main() {
    std::cout << "C++ Example of function" << std::endl;
    double x = 1.5;
    std::cout << x << " squared is " << f(x) << std::endl;
}
